flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -591,5 +729,27 @@ object AggregateUtil {
 
     groupingOffsetMapping.toArray
   }
+
+  private def getTimeFieldPosition(timeField: Expression, inputType: RelDataType): Int = {
+    timeField match {
+      case ResolvedFieldReference(name, resultType) =>
+        // get the RelDataType referenced by the time-field
+        val relDataType = inputType.getFieldList.filter(r => name.equals(r.getName))
+        // should only match one
+        if (relDataType.length == 1) {
+          relDataType.head.getIndex
+        } else {
+          throw new IllegalArgumentException()
```

Comment:
I would throw a general `TableException` and give a meaningful message here and in the following 2 exceptions.